packages/itwin/reports-config-widget/src/widget/components/BulkExtractor.ts

```diff
@@ -49,34 +56,75 @@ export default class BulkExtractor {
         if (state) {
           states.push(state);
         } else {
-          const runState = await this.getSingleState(runId, await this._accessToken());
+          const runState = await this.getState(runId, await this._accessToken());
           states.push(runState);
           stateByRunId.set(runId, runState);
         }
       }
       const finalState = BulkExtractor.getFinalState(states);
       stateByReportId.set(reportId, finalState);
     }
-    return stateByReportId;
-  }
 
-  private async fetchStates(): Promise<void> {
-    this._reportStates = await this.getStates(this._reportIds);
+    for (const iModelRun of this._iModelRun) {
+      let state = stateByRunId.get(iModelRun[1]);
+
+      if (!state) {
+        state = await this.getState(iModelRun[1], await this._accessToken());
+        stateByRunId.set(iModelRun[1], state);
+      }
+
+      const finalState = BulkExtractor.getFinalState([state]);
+      stateByIModelId.set(iModelRun[0], finalState);
+    }
+
+    this._reportStates = stateByReportId;
+    this._iModelStates = stateByIModelId;
   }
 
-  public getState(reportId: string): ExtractionStates {
+  public async getReportState(reportId: string): Promise<ExtractionStates> {
     if ((new Date().getTime() - this._timeFetched.getTime()) > STATUS_CHECK_INTERVAL) {
       this._timeFetched = new Date();
-      this.fetchStates().catch((e) =>
+      await this.fetchStates().catch((e) =>
```

Comment:
Change to `try/catch` instead of `.catch(() => ...)`

Reply:
Unnecessary `try/catch` removed